sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -297,6 +312,106 @@ export class AssessmentSequentialProcessingService {
     ];
   }
 
+  /**
+   * Gets the full-time program year contribution totals for the provided application numbers.
+   * @param applicationNumbers application numbers.
+   * @returns full-time program year contribution totals.
+   */
+  private async getProgramYearContributionTotals(
+    applicationNumbers: string[],
+  ): Promise<FTProgramYearContributionTotal[]> {
+    const totals = await this.applicationRepo
+      .createQueryBuilder("application")
+      .select(
+        "SUM((currentAssessment.workflowData -> 'calculatedData' ->> 'totalFederalFSC')::NUMERIC)",
+        FullTimeStudentContributionType.FederalFSC,
+      )
+      .addSelect(
+        "SUM((currentAssessment.workflowData -> 'calculatedData' ->> 'totalProvincialFSC')::NUMERIC)",
+        FullTimeStudentContributionType.ProvincialFSC,
+      )
+      .addSelect(
+        "SUM((currentAssessment.workflowData -> 'calculatedData' ->> 'exemptScholarshipsBursaries')::NUMERIC)",
+        FullTimeStudentContributionType.ScholarshipsBursaries,
+      )
+      .addSelect(
+        "SUM((currentAssessment.workflowData -> 'calculatedData' ->> 'studentSpouseContributionWeeks')::NUMERIC)",
+        FullTimeStudentContributionType.SpouseContributionWeeks,
+      )
+      .innerJoin("application.currentAssessment", "currentAssessment")
+      .where("application.applicationNumber IN (:...applicationNumbers)", {
+        applicationNumbers,
+      }) // Overwritten application can have awards associated with and they should not be considered.
+      .andWhere("application.applicationStatus != :overwrittenStatus", {
+        overwrittenStatus: ApplicationStatus.Overwritten,
+      })
+      // Check for assessment completed status to avoid retrieving any cancelation status.
+      .andWhere(
+        "currentAssessment.studentAssessmentStatus = :completedStudentAssessmentStatus",
+        {
+          completedStudentAssessmentStatus: StudentAssessmentStatus.Completed,
+        },
+      )
+      .getRawOne<{
+        [FullTimeStudentContributionType.FederalFSC]: string;
+        [FullTimeStudentContributionType.ProvincialFSC]: string;
+        [FullTimeStudentContributionType.ScholarshipsBursaries]: string;
+        [FullTimeStudentContributionType.SpouseContributionWeeks]: string;
+      }>();
+
+    return Object.keys(FullTimeStudentContributionType).map((key) => ({
+      contribution: key as FullTimeStudentContributionType,
+      total: +totals[key] || 0,
```

Comment:
Should we return 0 for values that would not exist?

Reply:
This is a scenario, when the total is returned for full-time with application numbers, so that instead of returning null (when the json currentAssessment.workflowData -> 'calculatedData' has null)it will return 0.